shotover-proxy/src/server.rs

```diff
@@ -459,6 +469,45 @@ impl<C: Codec + 'static> Handler<C> {
 
             debug!("Received raw message {:?}", messages);
 
+            {
+                let mut already_modified = false;
```

Comment:
Will clean this up and add it as a method on `ConnectionState` instead of directly in `server.rs`. I will also add a flag to transforms as to whether they need `ConnectionState` and then we can opt to not run this check if we don't need it.